### PR TITLE
 Handle empty tensors in the backend and frontend 

### PR DESCRIPTION
**Pull request by Kh4L**
- Handle empty tensors in the backend and frontend
- Add test cases in C++ and new backend_impl python tests

### CODE REVIEW

**Comment by Kh4L** [comment]
Build 689592

File: qa/L0_python-self-test/test.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ cd dali/test/python
 
 test_body() {
     # test code
+    nosetests --verbose test_backend_impl.py
```
[the remark this answers is not in the document]
Comment (Kh4L) [reply qa/L0_python-self-test/test.sh]: Oh right, forgot to add it, thanks!

[the remark this answers is not in the document]
Comment (Kh4L) [reply qa/L0_python-self-test/test.sh]: Done

**Comment by Kh4L** [comment]
Build 690208

File: dali/pipeline/data/buffer.h [hunk 2]
```diff
@@ -123,6 +123,9 @@ class Buffer {
    * the non-const version of the method, or calling 'set_type'.
    */
   inline void* raw_mutable_data() {
+    // Empty tensor
+    if (size_ == 0 && data_.get() == nullptr)
```
Comment (jantonguirao) [inline dali/pipeline/data/buffer.h]: just wondering. Isn't one of those conditions enough to know that it's empy?

File: dali/pipeline/data/buffer.h [hunk 3]
```diff
@@ -273,6 +279,13 @@ class Buffer {
   inline void ResizeHelper(Index new_size) {
     DALI_ENFORCE(new_size >= 0, "Input size less than zero not supported.");
 
+    if (new_size == 0) {
```
Comment (jantonguirao) [inline dali/pipeline/data/buffer.h]: what is the reason for this?

File: dali/pipeline/data/buffer.h (hunk 2 above)
Comment (Kh4L) [reply dali/pipeline/data/buffer.h]: You're right, done

Comment (JanuszL) [reply dali/pipeline/data/buffer.h]: It seems to be not done yet.

**Comment by Kh4L** [comment]
Build 691053

**Pull request merged**